### ISSUE TITLE
Incorrect kwarg, this should error or fail

### ISSUE THREAD

**Issue by vpuri3** [JuliaDiff/SparseDiffTools.jl#252]
@gaurav-arya does it error when FunctionOp gets a wrong keyword argument?

https://github.com/JuliaDiff/SparseDiffTools.jl/blob/9c458cdc90c9b362c6f1c22a72d89a3466ea2ea7/test/test_vecjac_products.jl#L58

**Comment by gaurav-arya** [comment]
No, in the current design it shouldn't, because operators specify what kwargs they want and we filter the rest away. There would be a clear error if an operator doesn't get a keyword argument that it needs though, e.g. due to a misspelling.
